std/json.d

```diff
@@ -359,6 +359,90 @@ struct JSONValue
         return type == JSONType.null_;
     }
 
+    /***
+     * Generic type value getter
+     * A convenience getter that returns this `JSONValue` as the specified D type.
+     * Note: only arithmetic, `bool`, `string`, `JSONValue` and `JSONValue[]` types are accepted
+     * Throws: `JSONException` if `T` cannot hold the contents of this `JSONValue`
+     */
+    @property inout(T) get(T)() inout const pure @safe
+    {
+        static if (is(T : string))
```

Comment:
This `is`-expression requires that T is convertible to a string, not that the stores string can be (implicitly) coverted into a string. This would e.g. match the following `S`:

```
struct S {
	string toString() { return ""; }
	alias toString this;
}
```

This should either require an exact match `is(T == string)` or `is(string : T)` to support user defined types